android/sdl_android/src/main/java/com/smartdevicelink/util/SdlAppInfo.java

```diff
@@ -122,11 +182,115 @@ public String toString() {
+                }
+                eventType = parser.next();
+            }
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return vehicleMakesList;
+    }
+
+    /**
+     * Check to see if a vehicle type is supported.
+     *
+     * @param supportedVehicleList the list of supported vehicle types.
+     * @param connectedVehicle the vehicle type to check.
+     * @return true if vehicle type is supported.
+     */
+    public static boolean checkIfVehicleSupported(List<VehicleType> supportedVehicleList, VehicleType connectedVehicle) {
+        if (supportedVehicleList == null || supportedVehicleList.isEmpty() || connectedVehicle == null || connectedVehicle.getStore().isEmpty()) {
+            return true;
+        }
+        if (supportedVehicleList.contains(connectedVehicle)) {
+            return true;
+        }
+        for (VehicleType supportedVehicle: supportedVehicleList) {
+            String supportedVehicleMake = supportedVehicle.getMake();
+            String connectedVehicleMake = connectedVehicle.getMake();
+            if (supportedVehicleMake != null && connectedVehicleMake != null && connectedVehicleMake.equalsIgnoreCase(supportedVehicleMake)) {
+                String supportedVehicleModel = supportedVehicle.getModel();
+                String connectedVehicleModel = connectedVehicle.getModel();
+                if (supportedVehicleModel != null && connectedVehicleModel != null) {
+                    if (connectedVehicleModel.equalsIgnoreCase(supportedVehicleModel)) {
+                        boolean ret = true;
+                        String supportedVehicleModelYear = supportedVehicle.getModelYear();
+                        String connectedVehicleModelYear = connectedVehicle.getModelYear();
+                        if (supportedVehicleModelYear != null && connectedVehicleModelYear != null) {
+                            ret = connectedVehicleModelYear.equalsIgnoreCase(supportedVehicleModelYear);
+                        }
+                        String supportedVehicleTrim = supportedVehicle.getTrim();
+                        String connectedVehicleTrim = connectedVehicle.getTrim();
+                        if (supportedVehicleTrim != null && connectedVehicleTrim != null) {
+                            ret &= connectedVehicleTrim.equalsIgnoreCase(supportedVehicleTrim);
+                        }
+                        if (ret) {
+                            return true;
+                        }
+                    }
+                }
+                else {
+                    // Return true if only make is defined and it matches
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Gets app's supported vehicle types.
+     *
+     * @return List<VehicleType> a list representing the list of supported vehicle types.
```

Comment:
@YaroslavLutsenko on second thought I think this fits better
```suggestion
     * @return List<VehicleType> a list of supported vehicle types.
```

Reply:
@JulianKast (d236ac8)